specification/applicationinsights/resource-manager/microsoft.insights/stable/2015-05-01/favorites_API.json

```diff
@@ -0,0 +1,403 @@
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/ApplicationInsightsComponentFavorite"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The favorite definition updated.",
+            "schema": {
+                "$ref": "#/definitions/ApplicationInsightsComponentFavorite"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "FavoriteList": {
+            "$ref": "./examples/FavoriteUpdate.json"
+          }
+        }
+      },
+      "delete": {
+        "description": "Remove a favorite that is associated to an Application Insights component.",
+        "operationId": "Favorite_Delete",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceNameParameter"
+          },
+          {
+            "$ref": "#/parameters/FavoriteIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The favorite has been successfully removed from the Application Insights component."
+          }
+        },
+        "x-ms-examples": {
+          "FavoriteList": {
+            "$ref": "./examples/FavoriteDelete.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "ApplicationInsightsComponentFavorite": {
+      "description": "Properties that define a favorite that is associated to an Application Insights component.",
+      "properties": {
+        "Name": {
+          "type": "string",
+          "description": "The user-defined name of the favorite."
+        },
+        "Config": {
+          "type": "string",
+          "description": "Configuration of this particular favorite, which are driven by the Azure portal UX. Configuration data is a string containing valid JSON"
+        },
+        "Version": {
+          "type": "string",
+          "description": "This instance's version of the data model. This can change as new features are added that can be marked favorite. Current examples include MetricsExplorer (ME) and Search."
+        },
+        "FavoriteId": {
+          "type": "string",
+          "description": "Internally assigned unique id of the favorite definition.",
+          "readOnly": true
+        },
+        "FavoriteType": {
+          "description": "Enum indicating if this favorite definition is owned by a specific user or is shared between all users with access to the Application Insights component.",
+          "type": "string",
+          "enum": [
+            "shared",
+            "user"
+          ],
+          "x-ms-enum": {
+            "name": "FavoriteType",
+            "modelAsString": false
```

Comment:
If an enum will ever have additional values, it should be modeled as string. If it will only have these values _forever_, then modelAsString false is appropriate. Please use this guidance for each of the enums.

Reply:
I will investigate the field and determine if it will ever be extended and alter it to `"modelAsString": true` if this is the case.

Reply:
Changed to modelAsString as we may need to add other favorite types in the future.

Reply:
@d3r3kk it sounds like you meant to set this to `true`? 🙂